### ISSUE TITLE
Group blocks won't validate after update to 6.0

### ISSUE THREAD

**Issue by nicpelletier** [WordPress/gutenberg#16328]
**Describe the bug**
After updating to 6.0, every group blocks show the validation error "This block contains unexpected or invalid content.". The "Resolve" function won't work. It converts the block to a Custom HTML. The console shows this :
Block validation: Block validation failed for `core/group` ({name: "core/group", save: ƒ, category: "layout", attributes: {…}, title: "Group", …}).
Content generated by `save` function:
```
<div class="wp-block-group strate accent" id="group-125"><div class="wp-block-group__inner-container"></div></div>
```
Content retrieved from post body:
```
<div class="wp-block-group strate accent" id="group-125">
</div>
```

**Desktop:**
 - OS: Windows 10, version 1703
 - Browser : Chrome
 - Version : 75.0.3770.100

**Additional context**
As you can see in the HTML outputed by the console, i have a custom function that adds an id to every group blocks. They also have custom classes added in the advanced tab ("strate" and "accent").

If someone have an idea on how i could fix this via the database it would be appreciated. All my site content is wrapped in a group. As for now, i can't do any modification to my site without loosing a page content...

Thanks, and sorry for the bad english...

**Comment by youknowriad** [comment]
@kjellr I think this is related to the inner container PR. I think the deprecated definition is probably not 100% accurate.

**Comment by kjellr** [comment]
Uh oh — that's unfortunate. @talldan, since you [built](https://github.com/WordPress/gutenberg/pull/15210/commits/aa1646282debdc0e1327219428b316b048771126) that into the PR, is this something you can take a look at?

**Comment by talldan** [comment]
Taking a look.

**Comment by talldan** [comment]
Hmm, either I've misunderstood, or it seems as though `supports: anchor` doesn't work for a deprecation. The headings block may have also had this issue.

**Comment by talldan** [comment]
😟Oh boy, I have some concern that I might have broken how `supports` works for deprecations in this PR:
https://github.com/WordPress/gutenberg/pull/12757

It seems like the filter for adding `supports` attributes doesn't run for deprecations. e.g.:
```
addFilter( 'blocks.registerBlockType', 'core/anchor/attribute', addAttribute );
```

Also seeing the same issues with `align` support, but that doesn't cause a validation issue as there's more leniency with data and class attributes.

**Comment by nicpelletier** [comment]
I confirm that all my headers alignments are gone as well.

**Comment by talldan** [comment]
Doesn't seem related to #12757, I tested reverting that commit and I'm seeing the same behaviour. It seems as though I'd have to declare the anchor attribute on the deprecation.

**Comment by talldan** [comment]
Fix at #16346.
